frontend/pages/queries/edit/components/EditQueryForm/EditQueryForm.tsx

```diff
@@ -610,25 +614,27 @@ const QueryForm = ({
 
   const hasSavePermissions = isGlobalAdmin || isGlobalMaintainer;
 
-  const hasSqlChange = storedQuery && lastEditedQueryBody !== storedQuery.query;
-  const hasSnapshotChange =
+  const currentlySavingQueryResults =
     storedQuery &&
-    lastEditedQueryLoggingType !== "snapshot" &&
-    storedQuery.logging === "snapshot";
-  // Use commented out logic when discard data checkbox is implemented #13470
-  const hasEnabledDiscardData = false;
-  // const hasEnabledDiscardData =
-  //   storedQuery && lastEditedDiscardData && !storedQuery.discardData;
-
-  const confirmChanges = (): boolean => {
-    // Confirm changes if the query has been edited, removed snapshot logging, or enabled discard data
-    return hasSqlChange || hasSnapshotChange || hasEnabledDiscardData;
-  };
+    !storedQuery.discard_data &&
+    !["differential", "differential_ignore_removals"].includes(
+      storedQuery.logging
+    );
+  const changedSQL = storedQuery && lastEditedQueryBody !== storedQuery.query;
+  const changedLoggingToDifferential = [
+    "differential",
+    "differential_ignore_removals",
+  ].includes(lastEditedQueryLoggingType);
 
-  const confirmSqlChange = (): boolean => {
-    // Confirm sql changes message if sql changed but snapshot and enabling discard data has not
-    return !!hasSqlChange && !hasSnapshotChange && !hasEnabledDiscardData;
-  };
+  const enabledDiscardData =
+    storedQuery && lastEditedQueryDiscardData && !storedQuery.discard_data;
+
+  const confirmChanges =
+    currentlySavingQueryResults &&
+    (!!changedSQL || !!changedLoggingToDifferential || !!enabledDiscardData);
```

Comment:
I think these are already booleans so you can remove  the `!!`, which will make it a little easier to read :)